### PR TITLE
Send letters to contacts with no email address

### PR DESCRIPTION
**Pull request by Beckyrose200**
https://eaflood.atlassian.net/browse/RUBY-1997

Here we have the changes to allow letters to be sent to registrations without an email address.
In WEX there are two emails address captured, applicant_email and contact_email.
These changes mean letters are sent to any emails that are nil value instead of the NCCC email address.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -204,6 +204,8 @@
     t.string "temp_reuse_address_for_site_location"
     t.boolean "temp_use_registered_company_details"
     t.datetime "companies_house_updated_at"
+    t.boolean "temp_reuse_applicant_name"
```
Comment (PaulDoyle-EA) [inline db/schema.rb]: Please check why this change crept in. Possibly related to ongoing issues around WEX migrations.

Comment (Beckyrose200) [reply db/schema.rb]: The `temp_reuse_applicant_name` change was merged into the engine on July 11th. The back office schema hasn't been updated with this change since the merge. I ran a db:migrate due to issues with my db so I think that's why this PR has the updated schema.

Comment (Beckyrose200) [reply db/schema.rb]: I think the front office needs the schema updating as well

File: app/controllers/resend_confirmation_email_controller.rb [hunk 2]
```diff
@@ -41,7 +41,7 @@ def send_emails
 
   def emails_to_contact
     emails = [registration.applicant_email, registration.contact_email]
-    emails.delete(WasteExemptionsEngine.configuration.assisted_digital_email)
+    emails.compact!
     emails.uniq
   end
```
Comment (d-a-v-e) [inline app/controllers/resend_confirmation_email_controller.rb:46]: looks like you can lose the assignment here, so you could just write (note the removal of the `!` in `compact!`):

```ruby
def emails_to_contact
  [registration.applicant_email, registration.contact_email]
    .delete(WasteExemptionsEngine.configuration.assisted_digital_email)
    .compact
    .uniq
end
```

File: app/services/remove_nccc_email_service.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+# rubocop:disable Layout/LineLength
+class RemoveNcccEmailService < ::WasteExemptionsEngine::BaseService
+
+  # https://rubular.com/r/LPHA79nxD5k2ff
```
Comment (d-a-v-e) [inline app/services/remove_nccc_email_service.rb]: this is really handy. (Make sure it is cleared on rubular.com once this is merged!)

File: app/services/remove_nccc_email_service.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    Rails.logger.info "Clearing wex applicant email addresses: #{matching_registrations.pluck(:applicant_email)}"
+    matching_registrations.update_all(applicant_email: nil)
+
+    matching_registrations = find_email_variations(:contact_email)
+
+    return unless matching_registrations.any?
+
+    Rails.logger.info "Clearing wex contact email addresses: #{matching_registrations.pluck(:contact_email)}"
+    matching_registrations.update_all(contact_email: nil)
+  end
```
Comment (d-a-v-e) [inline app/services/remove_nccc_email_service.rb:33]: So if there are no matches found for `:applicant_email` then the rest of the searches have no reason to run?

**Review by d-a-v-e — CHANGES_REQUESTED** [review]

File: app/services/remove_nccc_email_service.rb (hunk 4 above)
Comment (Beckyrose200) [reply app/services/remove_nccc_email_service.rb:33]: This was another annoying rubocop error. I had them in if statements and rubocop disagreed.

File: app/controllers/resend_confirmation_email_controller.rb (hunk 2 above)
Comment (Beckyrose200) [reply app/controllers/resend_confirmation_email_controller.rb:46]: That looks good thank you

File: app/models/waste_exemptions_engine/registration.rb [hunk 5]
```diff
@@ -18,7 +16,7 @@ class Registration < ::WasteExemptionsEngine::ApplicationRecord
 
     scope :renewals, -> { where.not(referring_registration_id: nil) }
 
-    scope :contact_email_is_not_nccc, -> { where.not(contact_email: NCCC_EMAIL) }
+    scope :contact_email_is_not_blank, -> { where.not(contact_email: nil) }
```
Comment (PaulDoyle-EA) [inline app/models/waste_exemptions_engine/registration.rb]: "blank" and "nil" are not the same - maybe rename this to `contact_email_present` and also check for empty string values?

File: app/services/bulk_notify_renewal_letters_service.rb [hunk 6]
```diff
@@ -24,7 +24,7 @@ def ad_expiring_registrations
     @_ad_expiring_registrations ||= lambda do
       WasteExemptionsEngine::Registration
         .order(:reference)
-        .where(contact_email: WasteExemptionsEngine.configuration.assisted_digital_email)
+        .where(contact_email: nil)
```
Comment (PaulDoyle-EA) [inline app/services/bulk_notify_renewal_letters_service.rb:27]: Might need to also check for empty strings

File: app/services/remove_nccc_email_service.rb (hunk 3 above)
Comment (PaulDoyle-EA) [reply app/services/remove_nccc_email_service.rb]: ... in which case the comment should be deleted ....

**Review by PaulDoyle-EA — APPROVED** [review]

**Review by d-a-v-e — APPROVED** [review]
